inference-engine/src/inference_engine/generic_ie.cpp

```diff
@@ -129,8 +129,9 @@ void ngraph::op::GenericIE::validate_and_infer_types() {
             }
         }
 
-        // WA: Proposal shape infer has to know number of outputs
-        if (type == "Proposal" && parameters.find("num_outputs") == parameters.end()) {
+        // WA: shape infer has to know number of outputs
+        if ((type == "Proposal" || type == "ExperimentalDetectronROIFeatureExtractor" || type == "ExperimentalDetectronDetectionOutput")
```

Comment:
We can get rid of workaround for Proposal since it now works through nGraph op.

Reply:
decided to leave it in case there is some backward compatibility dependency.